### PR TITLE
Added THETA support

### LINKED ISSUE #102: Add support for THETA addresses in encoding library
Here is the encoding library: https://github.com/ensdomains/address-encoder

Example: https://github.com/ensdomains/address-encoder/pull/47/files

Discord for dev help: https://discord.com/invite/AskZbFx

Please provide a reference link so we can check you did the encoding correctly.

We'd like to keep the encoding library as small as possible, so please be efficient in your use of space.

### PR DESCRIPTION
**Pull request by surajsingla333**
## Issue number

Closes #102 

## Description

The PR is related to the gitcoin bounty, https://gitcoin.co/issue/ensdomains/address-encoder/102/100023813, I was not able to start the work because simultaneously another user started it. The THETA token encoder and decoder is added and the test cases are also confirmed.

## List of features added/changed

- Added THETA support using ethereum chain encoder/decoder

## Reference

https://wallet.thetatoken.org/unlock/keystore-file

## How Has This Been Tested?

- Using ethereum encoder to generate hex format of THETA addressed

## Screenshots (if appropriate):

Screenshot (3154)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My code implements all the required features.

### CODE REVIEW

**Comment by surajsingla333** [comment]
Hey @Arachnid this branch is updated as per the new tests and is ready to be merged.

File: src/index.ts [hunk 1]
```diff
@@ -634,15 +634,15 @@ function algoDecode(data: string): Buffer {
...
-  // Calculate publicKey checksum
-  const checksum = algoChecksum(data);
-
-  // Append publicKey and checksum
-  const addr = b32encode(hex2a(data.toString('hex').concat(checksum)));
-
-  // Removing the extra '='
-  const cleanAddr = addr.replace(/=/g, '');
-  return cleanAddr;
+    // Calculate publicKey checksum
```
Comment (makoto) [inline src/index.ts:637]: Please remove extra whitespace

File: src/index.ts [hunk 2]
```diff
@@ -753,16 +753,17 @@ function aionEncoder(data: Buffer): string {
 }
 
 const getConfig = (name: string, coinType: number, encoder: EnCoder, decoder: DeCoder) => {
-  return {
-    coinType,
-    decoder,
-    encoder,
-    name,
-  };
+    return {
```
Comment (makoto) [inline src/index.ts:756]: Please remove extra whitespace

File: src/index.ts [hunk 3]
```diff
@@ -753,16 +753,17 @@ function aionEncoder(data: Buffer): string {
...
+        coinType,
+        decoder,
+        encoder,
+        name,
+    };
 };
 
 // Ordered by coinType
 export const formats: IFormat[] = [
+
```
Comment (makoto) [inline src/index.ts:766]: Please remove extra whitespace

**Comment by makoto** [comment]
`https://wallet.thetatoken.org/unlock/keystore-file` Please be more specific about where in the page it specifies that it's a hex with checksum? By looking at https://explorer.thetatoken.org/account/0x80eab22e27d4b94511f5906484369b868d6552d2 it doesn't even look like it has checksum

**Comment by makoto** [comment]
Hi, please let me know if you want to continue working on this or you want me to pass on to someone else.

**Pull request closed without merging**